src/app/vertextool/qgsvertextool.cpp

```diff
@@ -2430,28 +2432,28 @@ void QgsVertexTool::highlightAdjacentVertex( double offset )
   zoomToVertex( vertex );  // make sure the vertex is visible
 }
 
-void QgsVertexTool::startSelectionRect( QPoint point0 )
+void QgsVertexTool::initSelectionRubberBand()
```

Comment:
These changes seem unrelated -- can you defer these to a follow-up PR?

Reply:
Sorry, could you be more specific? Do you mean renaming the methods?

Reply:
I mean that the actual fix is just the change in cadCanvasReleaseEvent from using the rect to using the geometry (which is a great fix, thanks!). The rest of these changes are unrelated (e.g. why have you changed the colors of the rubber band?) Those are the bits which should be left to a separate PR for discussion.

Reply:
Well, in order to get the geometry I changed from using a QRubberBand to a QgsRubberBand, but it defaults to different colors so I tried to make it look just like before so the ux doesn't change (couldn't get the exact color QRubberBand uses by default, maybe it's system specific?).
Then the start-update-stop methods had to be adapted for mSelectionRubberBand.
After that, renaming the methods to reflect that they fiddle with the SelectionRubberBand seemed logical but obviously not necessary. The rest though I think are necessary because of the transition from QRubberBand to QgsRubberBand

Reply:
Thanks for the explanation! I missed the QRubberBand -> QgsRubberband swap here, I'll re-review accordingly